core/store/src/trie/mod.rs

```diff
@@ -1378,7 +1380,9 @@ impl Trie {
         let charge_gas_for_trie_node_access =
             mode == KeyLookupMode::Trie || self.charge_gas_for_trie_node_access;
         if self.memtries.is_some() {
-            self.lookup_from_memory(key, charge_gas_for_trie_node_access)
+            self.lookup_from_memory(key, charge_gas_for_trie_node_access, |v| {
```

Comment:
Seems like this is the main change we are striving for?

Reply:
Umm, on one hand yes, on the other introducing a callback was forced by the fact that `self.memtries` are behind a lock. I would have rathered something cleaner TBH.